projects/core/src/site-context/providers/context-service-providers.ts

```diff
@@ -5,6 +5,7 @@ import { CurrencyService } from '../facade/currency.service';
 import { LanguageService } from '../facade/language.service';
 import { SiteContextRoutesHandler } from '../services/site-context-routes-handler';
 
+// TODO 4.0: remove the deprecated initialization
```

Comment:
I hope you mean you plan to remove only?:
```ts
langService.initialize();
currService.initialize();
```

We still need to call:
```ts
siteContextRoutesHandler.init();
```
and I think also:
```ts
baseSiteService.initialize();
```

Please mind that your new [Langauge|Currency]StatePersistance app initializers should run after `siteContextRoutesHandler.init();` is called.

Currently I believe it's the case, becasue they are provided after that one app initializer above and `baseSiteService.initialize();` is synchronous.

But please double check with:
- static config
- auto-config

Reply:
Yes I meant only remove:
```
langService.initialize();
currService.initialize();
```